packages/essentials/utilities/timing/Timer.test.ts

```diff
@@ -149,6 +159,145 @@ describe("Timer", () => {
     });
   });
 
+  describe("Set duration", () => {
+    it("should change the timer duration and update the remaining time", () => {
+      timer.start();
+      timer.setDuration(2000);
+
+      expect(timer.progress).toBe(0);
+      expect(timer.duration).toBe(2000);
+      expect(timer.remaining).toBe(2000);
+      expect(timer.status).toBe(TimerStatus.running);
+      expect(callback).not.toHaveBeenCalled();
+
+      vi.advanceTimersByTime(2000);
+
+      expect(timer.status).toBe(TimerStatus.completed);
+      expect(callback).toHaveBeenCalled();
+    });
+
+    it("should update the timer duration and update the remaining time when paused", () => {
+      timer.start();
+      timer.pause();
+      timer.setDuration(7000);
+
+      expect(timer.duration).toBe(7000);
+      expect(timer.remaining).toBe(7000);
+    });
+
+    it("should extend a completed timer duration and show correct status", () => {
```

Comment:
Nitpicking but do we need to run linters for this file? 😅

Reply:
Already did, this is just looking weird because github is breaking it to the next line.